### DISCUSSION TITLE
Application.SetDefaultFont(Font) not available in VB.NET

### DISCUSSION THREAD

**Discussion by MichaelMayrhofer** [dotnet/winforms#5796] [Q&A] [answered]
* .NET Core Version: .NET 6.0.100-rc.1.21458.32

* Have you experienced this same bug with .NET Framework?: No

**Problem description:**
According to https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.application.setdefaultfont?view=net-6.0 there should be a method Application.SetDefaultFont(Font) also available in VB.NET, but it isn't. The compiler reports an error when I add this method call in my VB.NET app.

**Comment by RussKie** [comment, marked as answer]
The method is available, however it is available on [`System.Windows.Forms.Application` type](https://docs.microsoft.com/dotnet/api/system.windows.forms.application.setdefaultfont):
![image](https://user-images.githubusercontent.com/4403806/133739593-8c7bbcf6-335f-42fb-8cc5-b5b58e89b2ea.png)

VB has a property called `Application`, and this is causes the confusion.
![image](https://user-images.githubusercontent.com/4403806/133739912-914833de-0621-4e55-a00c-17bed3bc0f59.png)

**Reply by RussKie** [reply]
@KlausLoeffelmann can provide more guidance related to configuring default properties for VB apps.

**Reply by MichaelMayrhofer** [reply]
@RussKie Oh, I didn't notice that the property `Application` is in the `System.Windows.Forms` namespace. 
Thank you for your quick reply!

**Comment by DualBrain** [comment]
Was this possible collision not considered? Now that this one exists... what future ones may we expect? Would it make sense to alias (shadow or something) these in the VB side since they appear to be "the same thing"?

**Comment by KlausLoeffelmann** [comment]
Folks, this is VB! We will be OK!
This has been moved to the Application Framework, where it belongs.
So, starting with .NET 6, we have a new Application Event called `ApplyApplicationDefaults`.

Please take a look at the API Review discussion, why and how we had implemented it:
Also, as a FYI, we also fixed a bug where showing the Splash Screen through the App Framework always ate up 100% workload of a core. 
This has been fixed for .NET 6 as well.

https://github.com/dotnet/winforms/issues/5069

So, with this we're pretty well prepared to go forward, once we decide to tackle more defaults.
